### PR TITLE
Refactored Auth0 constructor and added tests

### PR DESCRIPTION
**Pull request by Nyholm**
Cleaned up the constructor. I did not change any logic, parameter or default values. I just made the code much easier to read and understand. 

This reduces the nPath complexity.

### CODE REVIEW

**Comment by cocojoe** [comment]
Would be a good opportunity to start adding tests.

**Comment by Nyholm** [comment]
Sure. Just give me a sec..

**Comment by cocojoe** [comment]
No rush

**Comment by Nyholm** [comment]
There you go, Some hot tests right from the oven.

File: tests/Auth0Test.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Auth0\Tests;
+
+use Auth0\SDK\Auth0;
+use Nyholm\NSA;
```
Comment (cocojoe) [inline tests/Auth0Test.php]: Hmm who wrotes this one ;) As this is just for tests this should be okay.

**Comment by Nyholm** [comment]
:)

Yeah. That lib should not be used for non-test code

**Comment by cocojoe** [comment]
Resolve and I'll merge.

**Comment by Nyholm** [comment]
Thank you. Resolved.

**Comment by Nyholm** [comment]
Thank you for merging.

**Pull request merged**